internal/service/rds/instance.go

```diff
@@ -1763,6 +1768,9 @@ func resourceInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 
 			if aws.StringValue(input.StorageType) == storageTypeIO1 {
 				input.Iops = aws.Int64(int64(d.Get("iops").(int)))
+			} else if aws.StringValue(input.StorageType) == storageTypeGP3 && aws.Int64(int64(d.Get("allocated_storage").(int))) >= 400 {
```

Comment:
This line doesn't compile.

Also, based on the documentation at https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Storage.html#Concepts.Storage.GeneralSSD, the storage threshold is `400 GiB` only for MySQL, PostgreSQL, and MariaDB. It is `200 GiB` for Oracle, and cannot be set for SQL Server.

Reply:
Don't try and check value here.